AutoDJ: make 'enable' shortcut work after startup

ronso0 opened the pull request:
I noticed two issue I need help with.
and I rather discuss them here to work around launchpad and ease code references.

**1)** before the AutoDJ hotkey Shift+F12 would toggle [AutoDJ],enable after startup any library feature must have had focus (Ctrl+F to focus the searchbar is sufficient, just ANY trackmodel needs to be selected?)

**2)** with 'Allow random track addition' disabled the above wouldn't work either:
- add some tracks to the AutoDJ queue, close Mixxx
- start Mixxx
- Ctrl+F to focus the searchbar
- press Shift + F12 to start AutoDJ
= log "Debug[Main] Queue is empty now"
from https://github.com/mixxxdj/mixxx/blob/2.3/src/library/autodj/autodjprocessor.cpp#L403
because getNextTrackFromQueue() returns 0 even though the queue holds enough tracks.
- click AutoDJ feature
- press Shift + F12 to start AutoDJ
= AutoDJ starts

**2** is solved with change added here: m_pAutoDJTableModel->select(); in autodjprocessor.
Though this is just a hack, I have no idea about the implications.

For **1** I have no idea..

ronso0 committed:
WIP allow to start AutoDJ without selecting the feature

daschuer commented:
A controller that has mapped [AutoDJ],enabled is also affected. 

I think that it is sufficient that m_pAutoDJTableModel->select(); is called once.

ronso0 commented:
what about

> 1) before the AutoDJ hotkey Shift+F12 would toggle [AutoDJ],enable after startup any library feature must have had focus

I doubt the change here fixes it for controllers as the shortcut is not working.
Where else do we need to m_pAutoDJTableModel->select(); ?

daschuer commented on src/library/autodj/autodjprocessor.cpp:
```diff
@@ -829,6 +829,7 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
 
 TrackPointer AutoDJProcessor::getNextTrackFromQueue() {
     // Get the track at the top of the playlist.
+    m_pAutoDJTableModel->select();
```
The Shift+F12 issue after start-up is also solved if this is moved here: 
https://github.com/mixxxdj/mixxx/blob/1ca77e00a53bbd63e5bb899ac27b8897efa95200/src/library/autodj/autodjprocessor.cpp#L126

daschuer replied:
And you need to add 
```
installEventFilter(m_pKeyboard);
```
here: 
https://github.com/mixxxdj/mixxx/blob/a74230c8c478e72e4d1af7ad0a3c0a2eb09388fe/src/mixxx.cpp#L164

ronso0 replied:
> ```
> mixxx.cpp#L164: installEventFilter(m_pKeyboard);
> ```

that did the trick! Thanks

ronso0 replied:
even though it's there, too https://github.com/mixxxdj/mixxx/blob/2.3/src/mixxx.cpp#L538

ronso0 committed:
AutoDJ: make 'enable' shortcut work after startup

Previously, one first had to focus a library widget and also enable
random track addition (even though the AutoDJ queue was populated).

ronso0 commented:
@daschuer Works beautifully, ready for review!

daschuer reviewed (approved):
LGTM, Thank you for taking care.
